core/src/main/scala/org/apache/spark/deploy/SparkSubmit.scala

```diff
@@ -444,6 +444,7 @@ object SparkSubmit {
       OptionAssigner(args.deployMode, ALL_CLUSTER_MGRS, ALL_DEPLOY_MODES,
         sysProp = "spark.submit.deployMode"),
       OptionAssigner(args.name, ALL_CLUSTER_MGRS, ALL_DEPLOY_MODES, sysProp = "spark.app.name"),
+      OptionAssigner(args.jars, ALL_CLUSTER_MGRS, CLIENT, sysProp = "spark.jars"),
```

Comment:
Thanks @vanzin @jerryshao for your input, some follow up questions:
- How should Spark behave when a user both sets `spark.jars` and `spark.yarn.dist.jars`? Do we want to set union them? (the comment on this method says "Set JAR files to distribute to the cluster")
- In `SparkConf` there is a `setJars()` method that sets `spark.jars`, should this method also set `spark.yarn.dist.jars` if we are in YARN mode?
- Does implementing option 2 resolve the issue completely? Because the repl main class needs the `spark.jar` system property to be set. Shouldn't we also set that system property somewhere?